icu4c/source/i18n/ulocdata.cpp

```diff
@@ -68,13 +68,16 @@ ulocdata_open(const char *localeID, UErrorCode *status)
 
    uld->noSubstitute = false;
    uld->bundle = ures_open(nullptr, localeID, status);
-   uld->langBundle = ures_open(U_ICUDATA_LANG, localeID, status);
 
    if (U_FAILURE(*status)) {
       uprv_free(uld);
       return nullptr;
    }
 
+   // ICU-22149: not all functions require lang data, so fail gracefully if it is not present
+   UErrorCode localStatus = U_ZERO_ERROR;
+   uld->langBundle = ures_open(U_ICUDATA_LANG, localeID, &localStatus);
```

Comment:
You don't need to check the actual error code here?  And we know that an error will mean `uld->langBundle` will sill be null?

Reply:
The error should _normally_ be missing-resource, but it could sometimes be memory-allocation or maybe something about the localeID.

Factory methods _should_ all return nullptr if they return with a failure code.